vowpalwabbit/global_data.cc

```diff
@@ -314,95 +314,96 @@ vw::vw()
   p = nullptr;
 
   reg_mode = 0;
-  current_pass = 0;
 
-  data_filename = "";
+  gs.current_pass = 0;
```

Comment:
These should be initialized in each subcomponents' constructor if at all possible. Dunno if that belongs in this PR or in a followup

Reply:
follow-up, feel free to unblock yourself if you're on top of this branch. We should get this in, to start chopping all those next tasks if not the PR will be to crazy to parse